app/code/Magento/PricingStorefront/Model/PriceManagement.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\PricingStorefront\Model;
+
+use Psr\Log\LoggerInterface;
+
+class PriceManagement
+{
+    /**
+     * @var PriceRepository
+     */
+    private $priceRepository;
+
+    /**
+     * @var PriceBookRepository
+     */
+    private $priceBookRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param PriceRepository $priceRepository
+     * @param PriceBookRepository $priceBookRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        PriceRepository $priceRepository,
+        PriceBookRepository $priceBookRepository,
+        LoggerInterface $logger
+    ) {
+        $this->priceRepository = $priceRepository;
+        $this->priceBookRepository = $priceBookRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Remove prices.
+     *
+     * @param string $priceBookId
+     * @param array $productIds
+     * @return int
+     * @throws \Throwable
+     */
+    public function unassignPrices(string $priceBookId, array $productIds): int
+    {
+        return $this->priceRepository->delete($priceBookId, $productIds);
+    }
+
+    /**
+     * Assign product price to price book.
+     *
+     * @param string $bookId
+     * @param array $price
+     * @param string|null $parentId
+     * @return array
+     */
+    public function assignPrice(string $bookId, array $price, ?string $parentId = null) : array
```

Comment:
Could you please describe in descriptionn the business logic of price assignment work?